hoomd/tune/attr_tuner.py

```diff
@@ -0,0 +1,332 @@
+        return self._get_target()
+
+    @target.setter
+    def target(self, value):
+        self._set_target(value)
+
+    @abstractmethod
+    def _get_x(self):
+        pass
+
+    @abstractmethod
+    def _set_x(self):
+        pass
+
+    @abstractmethod
+    def _get_y(self):
+        pass
+
+    def _get_target(self):
+        return self._target
+
+    def _set_target(self, value):
+        self._target = value
+
+    @property
+    def domain(self):
+        """tuple[any, any]: A tuple pair of the minimum and maximum accepted
+            values of x.
+
+            When, the domain is `None` then any value of x is accepted. Either
+            the minimum or maximum can be set to `None` as well which means
+            there is no maximum or minimum. The domain is used to wrap values
+            within the specified domain when setting x.
+        """
+        if self._domain is not None:
+            return tuple(self._domain)
+        else:
+            return None
+
+    @domain.setter
+    def domain(self, value):
+        if value is not None and not len(value) == 2:
+            raise ValueError("domain must be a sequence of length two.")
+        self._domain = value
+
+    def __hash__(self):
+        raise NotImplementedError("This object is not hashable.")
+
+    def __eq__(self, other):
+        raise NotImplementedError("This object is not equatable.")
+
+
+class ManualTuneDefinition(_TuneDefinition):
+    """
+    Class for defining y = f(x) relationships for tuning x for a set y target.
+
+    This class is made to be used with `hoomd.tune.SolverStep` subclasses.
+    Here y represents a dependent variable of x. In general, x and y should be
+    of type `float`, but specific `hoomd.tune.SolverStep` subclasses may accept
+    other types.
+
+    Args:
+        get_y (``callable``): A callable that gets the current value for y.
+        target (``any``): The target y value to approach.
+        get_x (``callable``): A callable that gets the current value for x.
+        set_x (``callable``): A callable that sets the current value for x.
+        domain (:obj:`tuple` [``any``, ``any``], optional): A tuple pair of the
+            minimum and maximum accepted values of x, defaults to `None`. When,
+            the domain is `None` then any value of x is accepted. Either the
+            minimum or maximum can be set to `None` as well which means there is
+            no maximum or minimum. The domain is used to wrap values within the
+            specified domain when setting x.
+    """
+    def __init__(self, get_y, target, get_x, set_x, domain=None):
+        self.__get_x = get_x
+        self.__set_x = set_x
+        self.__get_y = get_y
+        self._target = target
+        if domain is not None and not len(domain) == 2:
+            raise ValueError("domain must be a sequence of length two.")
+        self._domain = domain
+
+    def _get_x(self):
+        return self.__get_x()
```

Comment:
Is it standard Python convention to have two methods with the same name and different numbers of leading underscores? To avoid confusion, consider renaming `__get_x` to `_user_get_x` and similarly for the other attributes.

Reply:
I like that option better. I just didn't think of it when writing the code.